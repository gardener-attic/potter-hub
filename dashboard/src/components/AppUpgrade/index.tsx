import AppUpgrade from "./AppUpgrade";

export default AppUpgrade;
