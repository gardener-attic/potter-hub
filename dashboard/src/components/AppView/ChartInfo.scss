.ChartInfo > a {
  display: block;
  width: 100%;
}

.ChartInfoUpdate {
  color: white;
  width: 100%;
  margin-bottom: 0.5em;
  text-align: center;
  padding: 0.25em 0;
  font-size: 14px;
  background: #008145;
}

.ChartControllerManaged {
  text-align: center;
  background: rgba(9, 4, 80, 0.68);
  border-radius: 10px;
}
