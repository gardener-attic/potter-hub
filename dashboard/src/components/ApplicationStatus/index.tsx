import ApplicationStatus from "./ApplicationStatus";

export default ApplicationStatus;
