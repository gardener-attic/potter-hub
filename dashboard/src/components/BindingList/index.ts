import BindingList from "./BindingList";

export default BindingList;
