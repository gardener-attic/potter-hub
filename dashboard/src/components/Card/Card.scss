.Card,
.Card__inner {
  height: 100%;
  width: 100%;
}

.Card {
  margin: 0 1% 1.5em;
}

.Card a,
.Card a:hover {
  // TODO: See why we need important here - https://github.com/bitnami/hex/issues/217
  // sass-lint:disable no-important
  text-decoration: none !important;
}

.Card__Icon img {
  max-height: 64px;
}

/* max 4 items per row (default) */
@media screen and (min-width: 480px) {
  .Card.Card-responsive {
    width: 48%;
  }
}

@media screen and (min-width: 768px) {
  .Card.Card-responsive {
    width: 31%;
  }
}

@media screen and (min-width: 1100px) {
  .Card.Card-responsive {
    width: 23%;
  }
}

/* max 3 items per row */
@media screen and (min-width: 768px) {
  .Card.Card-responsive-3 {
    width: 48%;
  }
}

@media screen and (min-width: 1100px) {
  .Card.Card-responsive-3 {
    width: 31%;
  }
}

/* max 2 items per row */
@media screen and (min-width: 768px) {
  .Card.Card-responsive-2 {
    width: 48%;
  }
}
