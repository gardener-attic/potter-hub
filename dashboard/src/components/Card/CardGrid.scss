.CardGrid {
  display: flex;
  margin: 0 -1%;
  justify-content: space-between;
  flex-wrap: wrap;
}

.CardGrid:after {
  /* Display the last row properly: https://stackoverflow.com/a/34816625 */
  flex: auto;
  content: "";
}
