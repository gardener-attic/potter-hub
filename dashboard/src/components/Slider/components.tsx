// CODE FROM: https://sghall.github.io/react-compound-slider/#/slider-demos/horizontal
import * as React from "react";
import { GetHandleProps, GetTrackProps, SliderItem } from "react-compound-slider";

// *******************************************************
// HANDLE COMPONENT
// *******************************************************
interface IHandleProps {
  domain: number[];
  handle: SliderItem;
  getHandleProps: GetHandleProps;
}

export const Handle: React.SFC<IHandleProps> = ({
  domain: [min, max],
  handle: { id, value, percent },
  getHandleProps,
}) => (
  <div
    role="slider"
    aria-valuemin={min}
    aria-valuemax={max}
    aria-valuenow={value}
    style={{
      left: `${percent}%`,
      position: "absolute",
      marginLeft: "-11px",
      marginTop: "-6px",
      zIndex: 2,
      width: 24,
      height: 24,
      cursor: "pointer",
      borderRadius: "50%",
      boxShadow: "1px 1px 1px 1px rgba(0, 0, 0, 0.2)",
      backgroundColor: "#34568f",
    }}
    {...getHandleProps(id)}
  />
);

// *******************************************************
// TRACK COMPONENT
// *******************************************************
interface ITrackProps {
  source: SliderItem;
  target: SliderItem;
  getTrackProps: GetTrackProps;
}

export const Track: React.SFC<ITrackProps> = ({ source, target, getTrackProps }) => (
  <div
    style={{
      position: "absolute",
      height: 14,
      zIndex: 1,
      backgroundColor: "#7aa0c4",
      borderRadius: 7,
      cursor: "pointer",
      left: `${source.percent}%`,
      width: `${target.percent - source.percent}%`,
    }}
    {...getTrackProps()}
  />
);
