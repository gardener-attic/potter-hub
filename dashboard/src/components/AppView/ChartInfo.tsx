import * as React from "react";
import { AlertTriangle, CheckCircle } from "react-feather";
import { Link } from "react-router-dom";

import { IRelease } from "shared/types";
import { isAppManagedByBom } from "../../shared/utils";
import Card, { CardContent, CardFooter, CardGrid, CardIcon } from "../Card";

import placeholder from "../../img/placeholder.png";
import "./ChartInfo.css";

interface IChartInfoProps {
  app: IRelease;
}

class ChartInfo extends React.Component<IChartInfoProps> {
  public render() {
    const { app } = this.props;
    const name = app.name;
    const metadata = app.chart && app.chart.metadata;
    const icon = metadata && metadata.icon;
    const iconSrc = icon ? icon : placeholder;
    let notes = <span />;
    if (metadata) {
      notes = (
        <div>
          <div>Chart Name: {metadata.name}</div>
          <div>Chart Version: {metadata.version}</div>
          <div>
            <span>App Version: {metadata.appVersion || "-"}</span>
            {this.updateStatusInfo()}
          </div>
        </div>
      );
    }
    return (
      <CardGrid className="ChartInfo">
        <Card>
          <CardIcon icon={iconSrc} />
          <CardContent>
            <h5>{name}</h5>
            <p className="margin-b-reset">{metadata && metadata.description}</p>
          </CardContent>
          <CardFooter>
            <small>{notes}</small>
          </CardFooter>
        </Card>
      </CardGrid>
    );
  }

  private updateStatusInfo() {
    const { app } = this.props;
    // If update is not set yet we cannot know if there is
    // an update available or not
    if (app.updateInfo) {
      let updateContent: JSX.Element | null = null;
      if (app.updateInfo.error) {
        updateContent = (
          <React.Fragment>
            <AlertTriangle
              color="white"
              fill="#FDBA12"
              className="icon"
              size={15}
              style={{ bottom: "-0.2em" }}
            />{" "}
            <span>Update check failed. {app.updateInfo.error.message}</span>
          </React.Fragment>
        );
      } else {
        if (app.updateInfo.upToDate) {
          updateContent = (
            <React.Fragment>
              <CheckCircle
                color="#82C341"
                className="icon"
                size={15}
                style={{ bottom: "-0.2em" }}
              />{" "}
              Up to date
            </React.Fragment>
          );
        } else {
          const update = (
            <span>
              A new version is available:<br />
              - Chart Version: {app.updateInfo.chartLatestVersion}<br />
              - App Version: {app.updateInfo.appLatestVersion || "-"}
            </span>
          );
          updateContent = (
            <React.Fragment>
              <h5 className="ChartInfoUpdate">Update Available</h5>
              {update}
              <br />
              {!isAppManagedByBom(app.info?.Description) && (
                <span>
                  Click <Link to={`/apps/ns/${app.namespace}/upgrade/${app.name}`}>here</Link> to
                  upgrade.
                </span>
              )}
            </React.Fragment>
          );
        }
      }
      return (
        <div>
          <hr className="separator-small" />
          {updateContent}
        </div>
      );
    }
    return;
  }
}

export default ChartInfo;
