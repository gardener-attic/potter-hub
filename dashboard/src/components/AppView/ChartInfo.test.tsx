import { shallow } from "enzyme";
import context from "jest-plugin-context";
import * as React from "react";

import { hapi } from "shared/hapi/release";
import { IRelease } from "shared/types";
import { CardFooter } from "../../components/Card";
import ChartInfo from "./ChartInfo";

const defaultProps = {
  app: {
    chart: {
      metadata: {
        name: "bar",
        appVersion: "0.0.1",
        description: "test chart",
        icon: "icon.png",
        version: "1.0.0",
      },
    },
    name: "foo",
  } as hapi.release.Release,
};

it("renders a app item", () => {
  const wrapper = shallow(<ChartInfo {...defaultProps} />);
  expect(wrapper.find(".ChartInfo").exists()).toBe(true);
  expect(wrapper).toMatchSnapshot();
});

context("when information about updates is available", () => {
  it("renders an up to date message if there are no updates", () => {
    const appWithoutUpdates = { ...defaultProps.app, updateInfo: { upToDate: true } } as IRelease;
    const wrapper = shallow(<ChartInfo {...defaultProps} app={appWithoutUpdates} />);
    expect(wrapper.html()).toContain("Up to date");
  });
  it("renders an new version found message if the chart latest version is newer", () => {
    const appWithUpdates = {
      ...defaultProps.app,
      updateInfo: { upToDate: false, appLatestVersion: "0.0.1", chartLatestVersion: "1.0.0" },
    } as IRelease;
    const wrapper = shallow(<ChartInfo {...defaultProps} app={appWithUpdates} />);
    expect(
      wrapper
        .find(CardFooter)
        .children()
        .text(),
    ).toContain("A new version is available:- Chart Version: 1.0.0- App Version: 0.0.1");
  });
  it("renders an new version found message if the app latest version is newer", () => {
    const appWithUpdates = {
      ...defaultProps.app,
      updateInfo: { upToDate: false, appLatestVersion: "1.1.0", chartLatestVersion: "1.0.0" },
    } as IRelease;
    const wrapper = shallow(<ChartInfo {...defaultProps} app={appWithUpdates} />);
    expect(
      wrapper
        .find(CardFooter)
        .children()
        .text(),
    ).toContain("A new version is available:- Chart Version: 1.0.0- App Version: 1.1.0");
  });
  it("renders a warning if there are errors with the update info", () => {
    const appWithUpdates = {
      ...defaultProps.app,
      updateInfo: { error: new Error("Boom!"), upToDate: false, chartLatestVersion: "" },
    } as IRelease;
    const wrapper = shallow(<ChartInfo {...defaultProps} app={appWithUpdates} />);
    expect(wrapper.html()).toContain("Update check failed. Boom!");
  });
});
