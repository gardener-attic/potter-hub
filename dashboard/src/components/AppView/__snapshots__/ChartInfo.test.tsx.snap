// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a app item 1`] = `
<CardGrid
  className="ChartInfo"
>
  <Card>
    <CardIcon
      icon="icon.png"
    />
    <CardContent>
      <h5>
        foo
      </h5>
      <p
        className="margin-b-reset"
      >
        test chart
      </p>
    </CardContent>
    <CardFooter>
      <small>
        <div>
          <div>
            Chart Name: 
            bar
          </div>
          <div>
            Chart Version: 
            1.0.0
          </div>
          <div>
            <span>
              App Version: 
              0.0.1
            </span>
          </div>
        </div>
      </small>
    </CardFooter>
  </Card>
</CardGrid>
`;
